src/pool/spawn.rs

```diff
@@ -120,6 +129,68 @@ impl<T> QueueCore<T> {
             }
         }
     }
+
+    /// Scale workers.
+    pub fn scale_workers(&self, new_thread_count: usize) -> bool {
+        let _l = self.l.lock().unwrap();
+        if new_thread_count < self.config.min_thread_count
+            || new_thread_count > self.config.max_thread_count
+        {
+            return false;
+        }
+
+        let current_thread_count = self
+            .config
+            .core_thread_count
+            .swap(new_thread_count, Ordering::SeqCst);
+        let gap = new_thread_count as isize - current_thread_count as isize;
+        if gap > 0 {
+            let addr = self as *const QueueCore<T> as usize + 1;
+            for _ in 0..gap {
+                unsafe {
+                    parking_lot_core::unpark_one(addr, |_| UnparkToken(0));
+                }
+            }
+            self.core_workers.store(new_thread_count, Ordering::SeqCst);
+        } else {
+            let addr = self as *const QueueCore<T> as usize;
+            unsafe {
+                parking_lot_core::unpark_all(addr, UnparkToken(0));
+            }
+        }
+        true
+    }
+
+    pub fn need_to_park_worker(&self) -> bool {
+        let mut core_workers = self.core_workers.load(Ordering::SeqCst);
+        let core_thread_count = self.config.core_thread_count.load(Ordering::SeqCst);
+        if core_workers <= core_thread_count {
+            return false;
+        }
+        loop {
+            match self.core_workers.compare_exchange_weak(
+                core_workers,
+                core_workers - 1,
+                Ordering::SeqCst,
+                Ordering::SeqCst,
+            ) {
+                Ok(n) => {
+                    if n > core_thread_count {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+                Err(n) => {
+                    if n == core_thread_count {
```

Comment:
```suggestion
                    if n <= core_thread_count {
```
`core_thread_count` might be scaled down, and now it is a outdated value.